Adds a Maliput Railcar Model.

liangfok opened the pull request:
The content of this PR was extracted from #5377, which had grown too large.

# Visualization of the Flat Curved Lane

This PR includes a file called `flat_curved_lane.yaml` that is used to test the railcar. Here is a visualization of the `monolane::RoadGeometry` it specifies:

<img width="682" alt="screen shot 2017-03-05 at 8 43 47 pm" src="https://cloud.githubusercontent.com/assets/1388098/23625030/bb901b4c-0275-11e7-9138-c16661779762.png">

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/5419)
<!-- Reviewable:end -->

liangfok commented:
+@jadecastro for feature review please.
+@jwnimmer-tri for car review please.
Thanks!

(I will assign a platform reviewer after this PR passes feature review.)

---

Review status: 0 of 22 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-Ke_-wzzNZ7HcPmd5ny1:bqdqk1l)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Checkpoint.

I will put on the platform review hat as well as automotive hat.

---

Reviewed 19 of 22 files at r1.
Review status: 19 of 22 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/automotive/BUILD, line 226 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_1bX91g3AlQrTLAg5:-Ke_1bX91g3AlQrTLAg6:b4ci1uv) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/BUILD#L226)):*
> ```Python
>         ":generated_translators",
>         ":generated_vectors",
>         ":maliput_railcar",
> ```

I think this is not yet needed here?

---

*[drake/automotive/BUILD, line 534 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_1h0EOmp77RVZaNK_:-Ke_1h0EOmp77RVZaNKa:bm735ug) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/BUILD#L534)):*
> ```Python
> drake_cc_googletest(
>     name = "maliput_railcar_test",
>     srcs = ["test/maliput_railcar_test.cc"],
> ```

BTW This can be omitted (its the default).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-Ke_3vU6nKoE97e-DvAI:b-7k6ecj)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 2 of 22 files at r1.
Review status: 21 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_7vgG4S7GJ7j0YgLc:-Ke_7vgG4S7GJ7j0YgLd:b-v9x8d) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L64)):*
> ```cpp
>       context.get_continuous_state_vector();
>   const MaliputRailcarState<T>* const state =
>       dynamic_cast<const MaliputRailcarState<T>*>(&context_state);
> ```

FYI _Meta_ If you look at David's recent changes to SimpleCar (IIRC one on master, one in the velocity PR), there is a slightly shorter phrasing of how to obtain the state.  (Just computing `context_state` in the parens, without giving it a named local.)

---

*[drake/automotive/maliput_railcar.cc, line 67 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_8IPeV_cNEG5miSNG:-Ke_8IPfMCuQhwffhK6u:b-eg9whc) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L67)):*
> ```cpp
>   DRAKE_ASSERT(state != nullptr);
> 
>   // Obtains the output vector.
> ```

BTW output _vectors_ (plural).

BTW The names `output_vector` and `pose` could perhaps be more descriptive / have more parallel phrasing.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Ke_A1_llY2ghVsv47oj:b8jz7p2) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
> ```cpp
>       math::rpy2quat(Vector3<T>(rotation.roll, rotation.pitch, rotation.yaw));
>   pose->set_rotation(Eigen::Quaternion<T>(
>       quaternion(0), quaternion(1), quaternion(2), quaternion(3)));
> ```

BTW @liangfok Can you quickly ping dynamics team to make sure the prior four lines are the best-practice way to convert rpy into PoseVector::rotation, if you haven't already?

---

*[drake/automotive/maliput_railcar.cc, line 139 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_9K0CJ2g1XfsJIV8m:-Ke_9K0DSCPs9vHSYOs7:bpmjial) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L139)):*
> ```cpp
>     MaliputRailcarState<T>* rates) const {
>   const T speed = config.initial_speed();
>   if (state.s() >= lane_.length()) {
> ```

It seems like negative `s` should also use this `0` case.

---

*[drake/automotive/maliput_railcar.h, line 5 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_4taMPDBybXx5pph_:-Ke_4taMPDBybXx5ppha:b-ciom25) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.h#L5)):*
> ```objc
> #include <memory>
> 
> #include "drake/automotive/gen/driving_command.h"
> ```

Unused.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 18 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/automotive/BUILD, line 226 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_1bX91g3AlQrTLAg5:-Ke_G3-zzyTaB-uV4Gbc:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/BUILD#L226)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

I think this is not yet needed here?
</blockquote></details>

Done.

---

*[drake/automotive/BUILD, line 534 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_1h0EOmp77RVZaNK_:-Ke_G6kbiKbheS_h_DXi:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/BUILD#L534)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW This can be omitted (its the default).
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar.cc, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_7vgG4S7GJ7j0YgLc:-Ke_HAsLNDV48uQ7mqeY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L64)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

FYI _Meta_ If you look at David's recent changes to SimpleCar (IIRC one on master, one in the velocity PR), there is a slightly shorter phrasing of how to obtain the state.  (Just computing `context_state` in the parens, without giving it a named local.)
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar.cc, line 67 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_8IPeV_cNEG5miSNG:-Ke_HQcOHOXuxNyyvVC0:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L67)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW output _vectors_ (plural).

BTW The names `output_vector` and `pose` could perhaps be more descriptive / have more parallel phrasing.
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Ke_I7Q4_e7RieFuwVhX:b-6bxwdh) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW @liangfok Can you quickly ping dynamics team to make sure the prior four lines are the best-practice way to convert rpy into PoseVector::rotation, if you haven't already?
</blockquote></details>

Yes, I am currently awaiting a response.

---

*[drake/automotive/maliput_railcar.cc, line 139 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_9K0CJ2g1XfsJIV8m:-Ke_HjcdH3xsRNa3BD0B:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L139)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

It seems like negative `s` should also use this `0` case.
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar.h, line 5 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_4taMPDBybXx5pph_:-Ke_HqIQ1qOMY8uum5rl:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.h#L5)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Unused.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Review status: 18 of 22 files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Ke_L_o1RZ2feaY2V4Kq:bxnftsa) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Yes, I am currently awaiting a response.
</blockquote></details>

That looks correct to me. cc'ing @edrumwri, @hongkai-dai and @mitiguy since they recently made improvements to these quaternion conversion methods.
Also @hongkai-dai, why do we have that TODO in rpy2quat? is there a related issue number tracking it's resolution?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 5 of 5 files at r2.
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Ke_liRcWZk0yXiGisrW:b-k6w9w8) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

That looks correct to me. cc'ing @edrumwri, @hongkai-dai and @mitiguy since they recently made improvements to these quaternion conversion methods.
Also @hongkai-dai, why do we have that TODO in rpy2quat? is there a related issue number tracking it's resolution?

</blockquote></details>

It seems brittle to me that we have to call `rpt2quat` and then read out `0..3` manually to populate an `Eigen::Quaternion`; surely we have a function that does that for us?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KebR-fZmFM9D5VIfIFk:bjxcyfx) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

It seems brittle to me that we have to call `rpt2quat` and then read out `0..3` manually to populate an `Eigen::Quaternion`; surely we have a function that does that for us?
</blockquote></details>

@amcastro-tri , Eigen also implemented its Euler angles. Their implementation has some bugs. I sent them a patch and the problem was fixed. But we have not completely switched to Eigen's Euler angles yet. I think it is good to use `rpy2quat` here. Eigen's Euler angles is still in the unsupported modules, and they have not tested Euler angles with `T` being `autodiffscalar` yet.

The plan we talked before is to use Eigen's Euler angles, quaternions, angle axis and Isometry3d. But this change never happens. We basically do not have the man power to work on this change. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KebmwiQ6cDTXa05QyJd:b-b6h49b) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, hongkai-dai (Hongkai Dai) wrote…</i></summary><blockquote>

@amcastro-tri , Eigen also implemented its Euler angles. Their implementation has some bugs. I sent them a patch and the problem was fixed. But we have not completely switched to Eigen's Euler angles yet. I think it is good to use `rpy2quat` here. Eigen's Euler angles is still in the unsupported modules, and they have not tested Euler angles with `T` being `autodiffscalar` yet.

The plan we talked before is to use Eigen's Euler angles, quaternions, angle axis and Isometry3d. But this change never happens. We basically do not have the man power to work on this change. 
</blockquote></details>

Looks like the switch was discussed in #3470. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Kebo-ZqCWQ-StsTULgV:bcqpu3l) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Looks like the switch was discussed in #3470. 
</blockquote></details>

Thanks @liangfok , yes we discussed the issue, but we did not have a concrete plan on how and when to switch to Eigen. 

My personally feeling is that the code in Eigen Quaternion, AngleAxis and Isometry3d are very well tested. The EulerAngles, on the other hand, has not been tested as thoroughly as the other Eigen modules, so it stays in the unsupported package. I have seen in Drake that many people uses`Eigen::Quaternion`, `Eigen::AngleAxis` and `Eigen::Isometry3` directly, which I think is a good practice and we should continue to use these Eigen classes. The only exception is probably Euler angles. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Ked1LUCEk0X_oFhbeki:b-3afcvq) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, hongkai-dai (Hongkai Dai) wrote…</i></summary><blockquote>

Thanks @liangfok , yes we discussed the issue, but we did not have a concrete plan on how and when to switch to Eigen. 

My personally feeling is that the code in Eigen Quaternion, AngleAxis and Isometry3d are very well tested. The EulerAngles, on the other hand, has not been tested as thoroughly as the other Eigen modules, so it stays in the unsupported package. I have seen in Drake that many people uses`Eigen::Quaternion`, `Eigen::AngleAxis` and `Eigen::Isometry3` directly, which I think is a good practice and we should continue to use these Eigen classes. The only exception is probably Euler angles. 
</blockquote></details>

@hongkai-dai, yes, that was my understanding but clearly it was you who had to confirm. Would you have a link to this patch? or is there an ticket in Eigen tracking this issue? I think it'd be a good idea if you posted it in #3470 so we don't lose track of it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-Ked5fXhnPxogCk51k4F:bp8cw40) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

@hongkai-dai, yes, that was my understanding but clearly it was you who had to confirm. Would you have a link to this patch? or is there an ticket in Eigen tracking this issue? I think it'd be a good idea if you posted it in #3470 so we don't lose track of it.
</blockquote></details>

Okay, this is going nowhere, let me try to be more specific.

We have an (r,p,y) tuple.  We want to get an Eigen::Quaternion so we can set the rotation in a PoseVector.

Right now, the code is two separate statements, the second of which has a magic "0 1 2 3" in it (why not 3 0 1 2 for example)?  Shouldn't we have a _single_ function that turns (r,p,y) into Quaternion, with a simple unit test, so that call sites don't have to type magic integers everywhere?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KedAUKqVboTt73PaMkp:b-ci83ac) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Okay, this is going nowhere, let me try to be more specific.

We have an (r,p,y) tuple.  We want to get an Eigen::Quaternion so we can set the rotation in a PoseVector.

Right now, the code is two separate statements, the second of which has a magic "0 1 2 3" in it (why not 3 0 1 2 for example)?  Shouldn't we have a _single_ function that turns (r,p,y) into Quaternion, with a simple unit test, so that call sites don't have to type magic integers everywhere?
</blockquote></details>

Yes, I believe having a single rpy-to-Eigen::Quaternion method is desirable. Searching through Drake's source code, this appears to be the first instance where this need exists. Should we wait for the rule-of-three or add the new method now (in a separate PR that is merged before this one)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KedXKzvmuTdF7kz68HL:b-6bnrsv) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Yes, I believe having a single rpy-to-Eigen::Quaternion method is desirable. Searching through Drake's source code, this appears to be the first instance where this need exists. Should we wait for the rule-of-three or add the new method now (in a separate PR that is merged before this one)?
</blockquote></details>

Ok, I will push one in a bit if that helps.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KedYJHoo53RokZ_zO0g:b-igoygj) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

Ok, I will push one in a bit if that helps.
</blockquote></details>

Thanks @amcastro-tri!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
I have read everything once, _except_ for the unit test.  I am going to wait until feature review does a pass there before I finish up.

---

Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KedYHxNepEIzkBk8uIu:b-oqs90g) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

Ok, I will push one in a bit if that helps.
</blockquote></details>

I have marked myself satisfied here for the purposes of this PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-KedYLNroh_XCUGDbOmx:b7wegxn)*
<!-- Sent from Reviewable.io -->

jadecastro commented:
Checkpoint. PTAL.

Based on my read of the Automotive Planner System Models document, I thought that `MaliputRoadCar` would accept acceleration as an input port, but it looks as though `s_ddot` is set to zero.  Is this intentional?  Right now, it seems as though the current implementation is dynamically equivalent to `TrajectoryCar`.

---

Reviewed 16 of 22 files at r1, 5 of 5 files at r2.
Review status: 21 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 143 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeduMW3XZLYJFuDezxy:-KeduMW3XZLYJFuDezxz:b-frwkho) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.cc#L143)):*
> ```cpp
>     rates->set_s(speed);
>   }
>   rates->set_s_dot(0);
> ```

BTW, Is there any reason why `s_dot` is a state here, since the speed never changes? It seems unnecessary in this use-case.

---

*[drake/automotive/maliput_railcar.h, line 19 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kedg6eBf7fK_dLc-CS6:-Kedg6eBf7fK_dLc-CS7:b-pcej4u) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L19)):*
> ```objc
> /// predetermined speed.
> ///
> /// Configuration:
> ```

BTW, consider adding a description here stating that these configuration parameters are of the format given in `MaliputRailcarConfig`.

---

*[drake/automotive/maliput_railcar.h, line 31 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KedybkSTvP13KtwpRMc:-KedybkSTvP13KtwpRMd:b-jhqoiu) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L31)):*
> ```objc
> /// <B>Input Port Accessors:</B>
> ///
> ///   - None.
> ```

Based on the document, I thought the input would be `v_dot`, so that `MaliputRoadCar` would interface with `IdmController`? 

---

*[drake/automotive/maliput_railcar.h, line 35 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KedgCpQ8Uv_qnwxyZSV:-KedgCpQ8Uv_qnwxyZSW:b-bqcjl4) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L35)):*
> ```objc
> /// <B>Output Port Accessors:</B>
> ///
> ///   - state_output(): Contains this system's state vector.
> ```

BTW, consider adding a description here (or above) stating that these states are of the format given in`MaliputRailcarState`.

---

*[drake/automotive/maliput_railcar.h, line 78 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kee-_daCrVNYptffkIk:-Kee-_daCrVNYptffkIl:b-be6t9q) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L78)):*
> ```objc
>   static void SetDefaultParameters(MaliputRailcarConfig<T>* config);
> 
>   // Additional methods.
> ```

BTW, this description is a bit vague. Consider `Getter methods for the output port descriptors.` 

---

*[drake/automotive/maliput_railcar_config.named_vector, line 10 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kedumvt57f1_91l6lPV:-Kedumvu8H6Rtgr68vln:b-kl6rcc) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar_config.named_vector#L10)):*
> ```
> }
> element {
>     name: "initial_speed"
> ```

BTW, based on my read of the `Railcar` documentation and code, it seems this speed is held fixed? If so, consider changing the name from `initial_speed` to `speed`. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-KedyD-VMuebIm5axoOQ:bngowj6)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Yes, this is a first spiral that assumes no input and a fixed speed. I plan on following up with another PR that takes as input a desired acceleration, which is used to set `s_ddot` to be non-zero values. It is true that the current spiral-1 implementation is dynamically equivalent to `TrajectoryCar`.

---

Review status: 21 of 22 files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-Kee26kPjolXTeOX1mFI:b-oh4asu)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 21 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 143 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeduMW3XZLYJFuDezxy:-Kee2ghCBY1KAph6MT39:bouh48q) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.cc#L143)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, Is there any reason why `s_dot` is a state here, since the speed never changes? It seems unnecessary in this use-case.
</blockquote></details>

I plan on adding support for variable speeds in a follow-up PR. Should I remove it from this PR?

---

*[drake/automotive/maliput_railcar.h, line 31 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KedybkSTvP13KtwpRMc:-Kee3DWm5rpxilUx6IW9:bg2lj52) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L31)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

Based on the document, I thought the input would be `v_dot`, so that `MaliputRoadCar` would interface with `IdmController`? 
</blockquote></details>

Correct. Should I include support for `v_dot` in this PR? I'm afraid it will make this PR too big.

---

*[drake/automotive/maliput_railcar_config.named_vector, line 10 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kedumvt57f1_91l6lPV:-Kee3VXUDJF6-UfhwSO-:b8gnykn) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar_config.named_vector#L10)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, based on my read of the `Railcar` documentation and code, it seems this speed is held fixed? If so, consider changing the name from `initial_speed` to `speed`. 
</blockquote></details>

I plan on adding support for variable speeds in a follow-up PR. Given this, should i still rename it to be "speed" in this PR?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Review status: 21 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 143 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeduMW3XZLYJFuDezxy:-Kee4IEVKR3WDUAdYaC0:b6b06vl) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.cc#L143)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

I plan on adding support for variable speeds in a follow-up PR. Should I remove it from this PR?
</blockquote></details>

Please leave `s_dot` here for this PR.  If the code is unclear to the point of being defective, then a `TODO` explaining the coming work will suffice.

---

*[drake/automotive/maliput_railcar.h, line 31 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KedybkSTvP13KtwpRMc:-Kee4C2Ff4ef-T880rcn:b-kcocfc) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L31)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Correct. Should I include support for `v_dot` in this PR? I'm afraid it will make this PR too big.
</blockquote></details>

This is the correct spiral; `v_dot` will be next.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jadecastro commented:
Great - thanks. I'm definitely fine with that if we need to land something simple first.  Could you please add a TODO somewhere that an input will be added?

---

Review status: 21 of 22 files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-Kee2jgV91GSJ372kFtY:bnld25h)*
<!-- Sent from Reviewable.io -->

jadecastro commented:
Review status: 21 of 22 files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 143 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeduMW3XZLYJFuDezxy:-Kee4vXxdDJ5E08MxQaf:bthdqrg) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.cc#L143)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Please leave `s_dot` here for this PR.  If the code is unclear to the point of being defective, then a `TODO` explaining the coming work will suffice.
</blockquote></details>

Based on my f2f with Liang, I understand now why it's here, so it's fine leave it in.

---

*[drake/automotive/maliput_railcar_config.named_vector, line 10 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kedumvt57f1_91l6lPV:-Kee5EbPuQsf4DlXihoC:b-2ma4w7) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar_config.named_vector#L10)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

I plan on adding support for variable speeds in a follow-up PR. Given this, should i still rename it to be "speed" in this PR?
</blockquote></details>

Based on f2f, it's fine to keep it as is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

liangfok commented:
I addressed all reviewer comments. PTAL. Thanks.

---

Review status: 17 of 22 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/automotive/maliput_railcar.cc, line 143 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeduMW3XZLYJFuDezxy:-Kee5skfe9UhNmOgoAHz:b-gmbzfq) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.cc#L143)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

Based on my f2f with Liang, I understand now why it's here, so it's fine leave it in.
</blockquote></details>

Done. (Added TODO).

---

*[drake/automotive/maliput_railcar.h, line 19 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kedg6eBf7fK_dLc-CS6:-Kee69nEGJTHW47MV0kt:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L19)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, consider adding a description here stating that these configuration parameters are of the format given in `MaliputRailcarConfig`.
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar.h, line 35 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KedgCpQ8Uv_qnwxyZSV:-Kee6F5Jqit4hwYm6fww:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L35)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, consider adding a description here (or above) stating that these states are of the format given in`MaliputRailcarState`.
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar.h, line 78 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kee-_daCrVNYptffkIk:-Kee6azHmFzphbKby3un:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar.h#L78)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, this description is a bit vague. Consider `Getter methods for the output port descriptors.` 
</blockquote></details>

Done.

---

*[drake/automotive/maliput_railcar_config.named_vector, line 10 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Kedumvt57f1_91l6lPV:-Kee6crFByCT3VkUSCVX:boyexbg) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/maliput_railcar_config.named_vector#L10)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

Based on f2f, it's fine to keep it as is.
</blockquote></details>

Done. Thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-Kee6eWBW_941ZL7O1A1:bblnx7v)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 4 of 4 files at r3.
Review status: 21 of 22 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 22 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-KeeENSo9ZdE2uVNpMkV:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jadecastro commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, besides some minor comments. 

---

Reviewed 4 of 4 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/automotive/test/maliput_railcar_test.cc, line 96 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeeFTV2N4ZeVgp69qyD:-KeeFTV2N4ZeVgp69qyE:bkxcsje) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/test/maliput_railcar_test.cc#L96)):*
> ```cpp
>   // The arc radius of the road's s-axis when the road is created using
>   // InitializeCurvedMonoLane().
>   const double kCurvedRoadRadius{10};
> ```

BTW, Do you think this will be brittle to your `test/flat_curved_lane.yaml` file?  If so, is there an easy way to extract this directly from the generated `RoadGeometry` instead? Alternatively, a comment in the `yaml` file saying that it's linked to this file would be fine.

---

*[drake/automotive/test/maliput_railcar_test.cc, line 189 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeeGKwMZdZ-ejlGB8TY:-KeeGKwMZdZ-ejlGB8TZ:b-nk2w64) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/test/maliput_railcar_test.cc#L189)):*
> ```cpp
>   auto pose = pose_output();
>   Eigen::Isometry3d expected_pose = Eigen::Isometry3d::Identity();
>   expected_pose.translation() =  Eigen::Vector3d(0, 1.5, 8.2);
> ```

BTW (extra space before `Eigen`)

---

*[drake/automotive/test/maliput_railcar_test.cc, line 221 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeeGsOoBwyYov3gG-AM:-KeeGsOpqdXjUNzSQ0FU:b-5f733y) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/test/maliput_railcar_test.cc#L221)):*
> ```cpp
>   Eigen::Isometry3d expected_end_pose = Eigen::Isometry3d::Identity();
>   {
>     const Eigen::Vector3d rpy(0, 0, M_PI_2);
> ```

BTW, similar to above, `M_PI_2` (and perhaps some of the numbers below) may be brittle to the `.yaml` file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-:-KeeI_nx-6OnHoHkTzf6:b-dc130v)*
<!-- Sent from Reviewable.io -->

jadecastro commented:
Reviewed 1 of 22 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/automotive/test/maliput_railcar_test.cc, line 96 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeeFTV2N4ZeVgp69qyD:-KeeOYMFOm4V66dfXm1q:bh1u5nn) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/test/maliput_railcar_test.cc#L96)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, Do you think this will be brittle to your `test/flat_curved_lane.yaml` file?  If so, is there an easy way to extract this directly from the generated `RoadGeometry` instead? Alternatively, a comment in the `yaml` file saying that it's linked to this file would be fine.
</blockquote></details>

The information is stored in `drake::maliput::monolane::ArcLane` but is kept private with no accessor. @maddog-tri, would you be opposed to add a public `ArcLane::r()`?

Meanwhile, I have thus added documentation comments indicating the link.

---

*[drake/automotive/test/maliput_railcar_test.cc, line 189 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeeGKwMZdZ-ejlGB8TY:-KeePJU_jpzAz-NFsTl1:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/test/maliput_railcar_test.cc#L189)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW (extra space before `Eigen`)
</blockquote></details>

Done.

---

*[drake/automotive/test/maliput_railcar_test.cc, line 221 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-KeeGsOoBwyYov3gG-AM:-KeeQGlr38n-KVqx-JB4:bqkablw) ([raw file](https://github.com/robotlocomotion/drake/blob/f3b25496a52f8e8b2319d17a3a174cd53feb66be/drake/automotive/test/maliput_railcar_test.cc#L221)):*
<details><summary><i>Previously, jadecastro (Jonathan DeCastro) wrote…</i></summary><blockquote>

BTW, similar to above, `M_PI_2` (and perhaps some of the numbers below) may be brittle to the `.yaml` file.
</blockquote></details>

Done. Added documentation in the YAML file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

jadecastro commented:
Reviewed 2 of 2 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[drake/automotive/maliput_railcar.cc, line 102 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5419#-Ke_A1_kMD2-jfoUFYUf:-KedknzdRhGb1bGVW2Iz:bqqb3jy) ([raw file](https://github.com/robotlocomotion/drake/blob/37dada5dcd245f005e52b8c07bccdd223b8b8e74/drake/automotive/maliput_railcar.cc#L102)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

I have marked myself satisfied here for the purposes of this PR.
</blockquote></details>

@amcastro-tri , the patch has been merged by Eigen already. But Euler angles is still in the unsupported module.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5419)*
<!-- Sent from Reviewable.io -->
